### PR TITLE
[WIP] Metrics

### PR DESCRIPTION
**Pull request by ravigarg27**

### CODE REVIEW

File: pyglmnet/metrics.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+def deviance(y, yhat, ynull_, distr):
+
+	if distr in ['softplus', 'poisson']:
+		LS = utils.log_likelihood(y, y, distr)
+	else:
+		LS = 0
+
+	scores = list()
+	print "deviance"
```
Comment (jasmainak) [inline pyglmnet/metrics.py]: please remove the print statement

File: tests/test_metrics.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import numpy as np
```
Comment (jasmainak) [inline tests/test_metrics.py]: Author name missing

File: tests/test_pyglmnet.py [hunk 3]
```diff
@@ -222,6 +222,8 @@ def test_multinomial():
 
     # check that score is computed for all lambdas
     scorelist = glm_mn.score(X, y)
+    print scorelist.shape
+    print y_pred_proba.shape
```
Comment (jasmainak) [inline tests/test_pyglmnet.py]: can you use assert statements in test?

File: tests/test_metrics.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	glm_sim = GLM(score_metric='deviance')
+	X = np.random.randn(n_samples, n_features)
+	y = glm_sim.simulate(beta0, beta, X)
+
+	glm_sim.fit(X, y)
+	score = glm_sim[-1].score(X, y)
+	print score
+	assert_equal(score.shape[0], 1)
+
+def test_pseudoR2():
```
Comment (jasmainak) [inline tests/test_metrics.py]: missing docstring

File: pyglmnet/metrics.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		else :
+			yhat_this = yhat[idx, :, :]
+		
+		L1 = utils.log_likelihood(y, yhat_this, distr)
+		scores.append(-2 * (L1 - LS))
+	
+
+	return np.array(scores)
+
+def pseudo_R2(X, y, yhat, ynull_, distr):
```
Comment (jasmainak) [inline pyglmnet/metrics.py]: missing docstring

File: tests/test_metrics.py (hunk 2 above)
Comment (pavanramkumar) [inline tests/test_metrics.py]: irrelevant

File: tests/test_metrics.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    y = glm_sim.simulate(beta0, beta, X)
+
+    glm_sim.fit(X, y)
+    score = glm_sim[-1].score(X, y)
+
+    assert_equal(score.shape[0], 1)
+
+
+def test_accuracy():
+
```
Comment (pavanramkumar) [inline tests/test_metrics.py]: missing docstring

File: pyglmnet/metrics.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        if distr in ['softplus', 'poisson']:
+            scores.append(1 - (LS - L1) / (LS - L0))
+        else:
+            scores.append(1 - L1 / L0)
+
+    return np.array(scores)
+
+
+def accuracy(y, yhat):
+    """ Accuracy. """
```
Comment (pavanramkumar) [inline pyglmnet/metrics.py]: docstring incomplete. See pyglmnet methods like fit() or predict()

**Comment by pavanramkumar** [comment]
looks good! thanks

**Pull request merged**